### ISSUE TITLE
cmdLineTester_libpathTestRtfChild_0 segmentation fault on OpenJ9 JDK9 windows platform

### ISSUE THREAD

**Issue by vsebe** [eclipse/openj9#681]
```
===============================================
Running test cmdLineTester_libpathTestRtfChild_0 ...
===============================================
test with NoOptions
{ "C:/temp/bld_371236/jdk/bin\\java"  -Xcompressedrefs  -DJAVA_EXE="C:/temp/bld_371236/jdk/bin\\java" -DEXTRA_OPTIONS=" -Xcompressedrefs " -DJAVATEST_ROOT= -DJVM_TEST_ROOT=C:/temp/bld_371236/test///..//jvmtest \
-DTESTSJARPATH="C:/temp/bld_371236/test///..//jvmtest\\cmdLineTests\\libpathTest\\libpathTest.jar" \
-DRESJAR="C:/temp/bld_371236/test///..//jvmtest\\cmdline_options_testresources\\cmdlinetestresources.jar" -jar "C:/temp/bld_371236/test///..//jvmtest\\cmdline_options_tester\\cmdlinetester.jar" \
-config "C:/temp/bld_371236/test///..//jvmtest\\cmdLineTests\\libpathTest\\libpathRtfChild.xml" \
-nonZeroExitWhenError; \
if [ $? -eq 0 ] ; then echo "cmdLineTester_libpathTestRtfChild_0""_PASSED"; else echo "cmdLineTester_libpathTestRtfChild_0""_FAILED"; fi; } 2>&1 | tee -a TestTargetResult
*** Starting test suite: LIBPATH RtfChild test intended for PMR56610 (CMVC 201272) ***
Testing: RtfChild testcase
Test start time: 2017/11/23 22:06:27 Greenwich Mean Time
Running command: C:/temp/bld_371236/jdk/bin\java  -Xcompressedrefs  -Dcom.sun.management.jmxremote -cp C:/temp/bld_371236/test///..//jvmtest\cmdLineTests\libpathTest\libpathTest.jar org.openj9.test.libpath.RtfChild
Time spent starting: 32 milliseconds
Time spent executing: 1407 milliseconds
Test result: FAILED
Output from test:
 [ERR] Unhandled exception
 [ERR] Type=Segmentation error vmState=0x00040000
 [ERR] Windows_ExceptionCode=c00000fd J9Generic_Signal=00000004 ExceptionAddress=000007FEF2DB1D47 ContextFlags=0010005f
 [ERR] Handler1=000007FEF2D174E0 Handler2=000007FEF594AAF0
 [ERR] RDI=00000007FFFA9790 RSI=0000000000000000 RAX=000000000001A018 RBX=0000000001D613A0
 [ERR] RCX=0000000002316700 RDX=0000000002354C18 R8=0000000002354C18 R9=0000000002354D00
 [ERR] R10=0000000001D47000 R11=0000000001D53000 R12=00000007FFFA9790 R13=0000000000000000
 [ERR] R14=0000000002354D00 R15=00000007FFFA97E8
 [ERR] RIP=000007FEF2DB1D47 RSP=0000000001D612C0 RBP=0000000001D61500 GS=002B
 [ERR] FS=0053 ES=002B DS=002B
 [ERR] XMM0 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM1 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM2 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM3 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM4 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM5 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM6 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+000)
 [ERR] Module=C:\temp\bld_371236\jdk\bin\compressedrefs\j9vm29.dll
 [ERR] Module_base_address=000007FEF2AB0000 Offset_in_DLL=0000000000301d47
 [ERR] Target=2_90_20171123_371236 (Windows Server 2008 R2 6.1 build 7601 Sevice Pack 1)
 [ERR] CPU=amd64 (4 logical CPUs) (0x3fff7d000 RAM)
 [ERR] ----------- Stack Backtrace -----------
 [ERR] ---------------------------------------
 [ERR] JVMDUMP039I Processing dump event "gpf", detail "" at 2017/11/23 22:06:27 - please wait.
 [ERR] JVMDUMP032I JVM requested System dump using 'C:\temp\bld_371236\test\TestConfig\core.20171123.220627.5224.0001.dmp' in response to an event
 [ERR] JVMDUMP010I System dump written to C:\temp\bld_371236\test\TestConfig\core.20171123.220627.5224.0001.dmp
 [ERR] JVMDUMP032I JVM requested Java dump using 'C:\temp\bld_371236\test\TestConfig\javacore.20171123.220627.5224.0002.txt' in response to an event
 [ERR] JVMDUMP010I Java dump written to C:\temp\bld_371236\test\TestConfig\javacore.20171123.220627.5224.0002.txt
 [ERR] JVMDUMP032I JVM requested Snap dump using 'C:\temp\bld_371236\test\TestConfig\Snap.20171123.220627.5224.0003.trc' in response to an event
 [ERR] JVMDUMP010I Snap dump written to C:\temp\bld_371236\test\TestConfig\Snap.20171123.220627.5224.0003.trc
 [ERR] JVMDUMP007I JVM Requesting JIT dump using 'C:\temp\bld_371236\test\TestConfig\jitdump.20171123.220627.5224.0004.dmp'
 [ERR] JVMDUMP010I JIT dump written to C:\temp\bld_371236\test\TestConfig\jitdump.20171123.220627.5224.0004.dmp
 [ERR] JVMDUMP013I Processed dump event "gpf", detail "".
>> Success condition was not found: [Output match: I am the child]
>> Required condition was not found: [Output match: com.ibm.oti.vm.bootstrap.library.path=]
>> Required condition was not found: [Output match: sun.boot.library.path=]
>> Required condition was not found: [Output match: java.library.path=]
>> Failure condition was not found: [Output match: Could not find or load main class]
>> Failure condition was not found: [Output match: java.lang.UnsatisfiedLinkError]
>> Failure condition was not found: [Output match: Not found in com.ibm.oti.vm.bootstrap.library.path]
>> Failure condition was not found: [Output match: at Rtf.main]
>> Failure condition was not found: [Output match: at RtfChild.main]
>> Failure condition was found: [Output match: Unhandled Exception]
>> Failure condition was not found: [Output match: Exception:]
>> Failure condition was not found: [Output match: corrupt]
>> Failure condition was found: [Output match: Processing dump event]
>> Failure condition was found: [Output match: JVM requested Java dump]
>> Failure condition was found: [Output match: JVM requested Snap dump]

```

**Comment by smlambert** [comment]
Need to reinclude the test excluded in #697 when this issue is fixed.

**Comment by ChengJin01** [comment]
The trace/javacore indicates that this might be the same issue with issue #680, which is related to stack overflow.

**Comment by ChengJin01** [comment]
Already confirmed that this is an issue with growing stack when throwing out an exception of OS stack overflow, which hang was working on.

--------------------------

 stringObject: 00000007FFFAEAD0 loader: 00000007FFEF0000
04:10:00.360 0x19a00            j9vm.28   > sendLoadClass
04:10:00.376 0x19a00            j9vm.302  > **setCurrentExceptionUTF exceptionNumber=12 detailUTF=operating system stack overflow**
04:10:00.376 0x19a00            j9vm.294  > setCurrentException index=12 constructorIndex=0 detailMessage=00000007FFFAEB30
04:10:00.376 0x19a00            j9vm.50   > **Attempt to change stack size** from 18488 bytes to 1064960 bytes. SP=00000000000C9278, SOM=00000000000C5388, SOM2=00000000000C5388
04:10:00.376 0x19a00            j9vm.53   - Old stack = 00000000000C2B88, new stack = 00000000002140F8, delta = 000000000004A1A8 slots
04:10:00.376 0x19a00            j9vm.54   - Copying from 00000000000C9278 to 0000000000319FB8 for 0000000000000948 bytes
04:10:00.376 0x19a00            j9vm.75   - Update arg0EA at 0000000000319FF0 from 00000000000C92F0 to 000000000031A030
04:10:00.376 0x19a00            j9vm.75   - Update arg0EA at 000000000031A018 from 00000000000
...
04:10:00.376 0x19a00            j9vm.76   - Leaving arg0EA at 000000000031A8F8 as 0000000000000002
04:10:00.376 0x19a00            j9vm.57   - Update vmc SP to 0000000000319FB8, arg0EA to 000000000031A000
04:10:00.376 0x19a00            j9vm.60   - Old stack first has references in frame 00000000000C9BB8
04:10:00.376 0x19a00            j9vm.62   - Stack copied successfully
04:10:00.376 0x19a00            j9vm.63   < **Done growing stack**
04:10:00.376 0x19a00            j9vm.10   > internalSendExceptionConstructor
04:10:00.376 0x19a00            j9vm.179  > javaLookupMethod(vmStruct 0000000000019A00, targetClass 000000000003F500, nameAndSig 00007FF86A7406C8, senderClass 0000000000000000, lookupOptions 24624)
04:10:00.376 0x19a00            j9vm.180  - javaLookupMethod - methodName <init>
04:10:00.376 0x19a00            j9vm.538  - searching methods from 000000000003D5B0 using linear search
04:10:00.376 0x19a00            j9vm.181  < exit javaLookupMethod resultMethod 000000000003D5D0
Unhandled exception
Type=Segmentation error vmState=0x00040000
Windows_ExceptionCode=c00000fd J9Generic_Signal=00000004 ExceptionAddress=00007FF86A731D47 ContextFlags=0010005f
Handler1=00007FF86A697580 Handler2=00007FF87E71AAF0

**Comment by gacholio** [comment]
This is not a stack grow issue - throwing StackOverflowError always performs one extra stack grow to ensure there's enough stack to create the exception, so the grow being in the trace isn't indicative of a problem.

Adding "-Xiss1M -Xss1M" to the command line (initial stack size equal to maximum stack size disables the stack grower) does not fix the issue - still crashes in the same way.

**Comment by ChengJin01** [comment]
But the test works good if increasing the maximal stack size (e. g. specify "-Xiss2M -Xss2M"  or "-Xss2M" ). In such case, the stack grower should be still disabled.

**Comment by gacholio** [comment]
If you just increase the max (-Xss), does the test work?

**Comment by ChengJin01** [comment]
@gacholio , 
It always works only with -Xss (e.g. -Xss1090K) and no crash occurs.
In addition, it the maximal value has to be larger than 1090k in the test; otherwise it will throw out the following exception and no crash, which means the exception for OS stack overflow is thrown out correctly.

Exception in thread "main" java.lang.StackOverflowError: operating system stack overflow
        at java.lang.VMAccess.findClassOrNull(java.base@9-internal/Native Method)
        at java.lang.ClassLoader.loadClass(java.base@9-internal/ClassLoader.java:1020)
        at java.awt.font.TextAttribute.<clinit>(java.desktop@9-internal/TextAttribute.java:268)
        at sun.font.EAttribute.<clinit>(java.desktop@9-internal/EAttribute.java:45)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(java.base@9-internal/Native Method)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(java.base@9-internal/NativeMethodAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(java.base@9-internal/DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(java.base@9-internal/Method.java:564)
        at java.lang.Class.getEnumConstantsShared(java.base@9-internal/Class.java:2831)
        at java.lang.Class.getEnumConstants(java.base@9-internal/Class.java:2863)
        at sun.font.AttributeValues.<clinit>(java.desktop@9-internal/AttributeValues.java:214)
        at java.awt.Font.<clinit>(java.desktop@9-internal/Font.java:726)
        at sun.awt.windows.WComponentPeer.<clinit>(java.desktop@9-internal/WComponentPeer.java:569)
        at java.awt.Component.initIDs(java.desktop@9-internal/Native Method)
        at java.awt.Component.<clinit>(java.desktop@9-internal/Component.java:593)
        at sun.awt.windows.WToolkit.initIDs(java.desktop@9-internal/Native Method)
        at sun.awt.windows.WToolkit.<clinit>(java.desktop@9-internal/WToolkit.java:122)
        at java.lang.Class.forNameImpl(java.base@9-internal/Native Method)
        at java.lang.Class.forName(java.base@9-internal/Class.java:300)
        at java.awt.Toolkit$2.run(java.desktop@9-internal/Toolkit.java:573)
        at java.awt.Toolkit$2.run(java.desktop@9-internal/Toolkit.java:568)
        at java.security.AccessController.doPrivileged(java.base@9-internal/AccessController.java:638)
        at java.awt.Toolkit.getDefaultToolkit(java.desktop@9-internal/Toolkit.java:567)
        at java.awt.Toolkit.getEventQueue(java.desktop@9-internal/Toolkit.java:1457)
        at java.awt.EventQueue.isDispatchThread(java.desktop@9-internal/EventQueue.java:1050)
        at javax.swing.SwingUtilities.isEventDispatchThread(java.desktop@9-internal/SwingUtilities.java:1488)
        at javax.swing.text.StyleContext.reclaim(java.desktop@9-internal/StyleContext.java:456)
        at javax.swing.text.StyleContext.addAttribute(java.desktop@9-internal/StyleContext.java:313)
        at javax.swing.text.StyleContext$NamedStyle.addAttribute(java.desktop@9-internal/StyleContext.java:1539)
        at javax.swing.text.StyleContext$NamedStyle.setName(java.desktop@9-internal/StyleContext.java:1349)
        at javax.swing.text.StyleContext$NamedStyle.<init>(java.desktop@9-internal/StyleContext.java:1296)
        at javax.swing.text.StyleContext.addStyle(java.desktop@9-internal/StyleContext.java:108)
        at javax.swing.text.StyleContext.<init>(java.desktop@9-internal/StyleContext.java:88)
        at javax.swing.text.DefaultStyledDocument.<init>(java.desktop@9-internal/DefaultStyledDocument.java:108)
....

To sum up:
1) with -Xss1024k, crash occurs.
2) with -Xss > 1024K &  -Xss < 1090K,  it throws out the OS stack overflow exception above (should be considered as normal)
3) with -Xss >= 1090k,  it works good.
